pkg/storegateway/indexheader/thread.go

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+// Provenance-includes-location: https://github.com/cpg1111/threadpool-go/blob/master/thread.go
+// Provenance-includes-license: MIT
+// Provenance-includes-copyright: Christian Grabowski
+
+package indexheader
+
+import (
+	"runtime"
+
+	"go.uber.org/atomic"
+)
+
+type execResult struct {
+	value     interface{}
+	err       error
+	populated bool
```

Comment:
Who use this? I can't find in which condition would be `false`

Reply:
Exactly! It will never be `false` except for the zero value of this struct. When an `OSThread` is stopped the `res` channel is closed and so any further reads from it will result in the zero value for `execResult`. The code doing the read then checks for this to return the appropriate error when the channel is closed (and hence the thread pool is shutting down).